### PR TITLE
Easier session scoped mock (compatible with nested mocks)

### PR DESCRIPTION
**Pull request by clslgrnc**
Resolve conflicts between #116 and #110.

### CODE REVIEW

**Comment by clslgrnc** [comment]
f58c6844fa0652d9b129247ff5beea44292035b1 is necessary to make `real_http` work.
Indeed #116 changed call to `_original_send`, that always takes `session` as an argument, into a call to `self._last_send` for which `session` is implicit when mocking a specific session.

**Comment by clslgrnc** [comment]
There are still a few issues with nested global and session mocks like in the following example:
```python
    url = 'mock://example.test/'
    with requests_mock.Mocker() as global_mock:
        global_mock.get(url, text='global')
        session = requests.Session()
        with requests_mock.Mocker(session) as session_mock:
            global_mock.get(url, real_http=True)
            text = session.get(url).text
```

I guess `text` should be `"global"` since that global mock was in place at session creation.
Or maybe we can just state in the doc that global and session mocks should not be mixed up...

**Comment by clslgrnc** [comment]
Writing a proper test:
```python
def test_666():
    url = 'https://www.example.com/'
    with requests_mock.Mocker() as global_mock:
        global_mock.get(url, text='global')
        session = requests.Session()
        text = session.get(url).text
        assert text == 'global'
        with requests_mock.Mocker(session=session) as session_mock:
            session_mock.get(url, real_http=True)
            text = session.get(url).text
            assert text == 'global'
```
The first asset pass, but not the second one (the actual page from example.com is retrieved).
I'll have to investigate, but not today.

File: requests_mock/mocker.py [hunk 1]
```diff
@@ -68,6 +82,7 @@ class MockerCore(object):
     """
 
     def __init__(self, **kwargs):
+        self._mock_target = kwargs.pop('session', requests.Session)
```
[the remark this answers is not in the document]
Comment (jamielennox) [reply requests_mock/mocker.py]: Just going to fix this so we can merge and release :)

**Comment by jamielennox** [comment]
I did tack a small change on here just to clean it up so we can merge it rather than go back and forward in review. I hope this is ok for you both. 

Thanks you @clslgrnc @akeeman

**Pull request merged**